Remove the What's New modal and link

scruffian opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Fixes https://github.com/Automattic/wp-calypso/issues/100227

## Proposed Changes

* Removes the "What's New" modal and link in the Help Center.

## Why are these changes being made?

* This option wasn't being used enough to justify the code it needed.

## Testing Instructions
- Go to the help center
- Confirm you no longer see the What's New option

Also confirm that all the unused code is removed.

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->
Before
<img width="1196" alt="Screenshot 2025-03-20 at 18 03 04" src="https://github.com/user-attachments/assets/59285593-1e7c-493b-b8e9-0fb91a16719f" />


After:
<img width="587" alt="Screenshot 2025-03-27 at 11 35 29" src="https://github.com/user-attachments/assets/4eb088bc-e15c-4850-8b8b-e64de6ff0795" />


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

scruffian committed:
Remove the What's New modal and replace with a link to the blog

scruffian commented on packages/help-center/src/components/help-center-more-resources.tsx:
```diff
@@ -117,31 +71,19 @@ export const HelpCenterMoreResources = () => {
-							>
-								<Icon icon={ <NewReleases /> } size={ 24 } />
-								<span>{ __( "What's New", __i18n_text_domain__ ) }</span>
-								{ showWhatsNewDot && (
```
This is a pretty neat feature, its a shame to lose it!

fditrapani replied:
Is that a notification for when we make updates?

scruffian replied:
yeah

fditrapani replied:
You're right. That is too bad to lose this. I never noticed it before to be honest. I wonder if it really works.

tyxla replied:
If it wasn't used very often, then I think we should be safe to remove it.

scruffian commented on packages/help-center/src/components/help-center-more-resources.tsx:
```diff
@@ -117,31 +71,19 @@ export const HelpCenterMoreResources = () => {
+						<Button
+							variant="link"
+							className="help-center-more-resources__new-releases"
+							href="https://wordpress.com/blog/category/product-features/"
```
Need to consider if this will go to the right language...

fditrapani replied:
Good point. That would be really nice if we could account for language.

scruffian replied:
The other blogs don't see to have that category - see https://wordpress.com/fr/blog/

fditrapani replied:
Thanks for confirming. What are your thoughts about only showing this to English folks vs. keeping it for everyone and taking them to the English blog?

scruffian replied:
If we only show this to English speaking users, what would we do for the non-English speakers? It seems like this change will degrade their experience.

tyxla replied:
I think that's the corresponding category in French for example: https://wordpress.com/fr/blog/category/fonctionnalites/

However, not sure if we've mapped those correctly like posts are mapped when switching the language from the switcher in the bottom.

fditrapani replied:
> If we only show this to English speaking users, what would we do for the non-English speakers? It seems like this change will degrade their experience.

Good point. I just looked at tracks events and out of ~6,000-9,000 views/day of the help centre, this modal gets viewed ~20 times a day. The usage is so low, I'm wondering if it even really matters. What do you think?

fditrapani replied:
When we look at language breakdowns, the vast majority are english with 1-2 views from other locals. I think showing it to English only is fine if we even want to keep it.

fditrapani reviewed (approved):
I tested this and it works as described. Thanks for the update Ben!

tyxla commented on packages/help-center/src/components/help-center-more-resources.tsx:
```diff
@@ -2,55 +2,17 @@
 /* eslint-disable wpcalypso/jsx-classname-namespace */
 import { recordTracksEvent } from '@automattic/calypso-analytics';
 import { localizeUrl } from '@automattic/i18n-utils';
-import WhatsNewGuide, { useWhatsNewAnnouncementsQuery } from '@automattic/whats-new';
```
I wish we could remove the entire package! I think it's still used in `mu-wpcom-plugin`, but not sure if that's still the case and if it's still necessary.

tyxla commented on packages/help-center/src/components/help-center-more-resources.tsx:
```diff
@@ -2,55 +2,17 @@
-
-	const { hasSeenWhatsNewModal, doneLoading } = useSelect(
-		( select ) => ( {
-			hasSeenWhatsNewModal: (
```
Does it also make sense to change the help icon? Seems like we're still changing it when there are new items detected (and we're still doing those queries):

https://github.com/Automattic/wp-calypso/blob/0b0ff0a24ccd22b2cee57c73ac4a2830cf5e1287/packages/help-center/src/components/help-icon.tsx#L11-L30

scruffian replied:
oh yes, great catch, thanks

scruffian committed:
remove What's new and resize the panel

scruffian commented:
I updated this to totally remove the option and also made the panel smaller to compensate:

<img width="475" alt="Screenshot 2025-03-24 at 14 05 20" src="https://github.com/user-attachments/assets/40ee547c-af93-431e-a6da-e0f300c199ed" />

I'd like to find out where the apps version of this is loaded too...

tyxla commented:
> I'd like to find out where the apps version of this is loaded too...

Like I [mentioned earlier](https://github.com/Automattic/wp-calypso/pull/101692/files#r2006247066), it's used in `jetpack-mu-wpcom`, here:

https://github.com/Automattic/jetpack/blob/d49e2d38837accdf35f9d73cc2ade742b1393d3f/projects/packages/jetpack-mu-wpcom/src/features/wpcom-whats-new/wpcom-whats-new.php#L47

scruffian commented:
Studio are planning to add a feature like this which pops up when a change happens in the app. Should we consider making the modal more intrusive and appearing when things change, so that users see it, like they are doing in Studio?

fditrapani commented:
> Studio are planning to add a feature like this which pops up when a change happens in the app. Should we consider making the modal more intrusive and appearing when things change, so that users see it, like they are doing in Studio?

I think it's best we take this out for now and revaluate our approach if we want to add something like this again.

I'd advise against the modal approach. I've sat in on enough usability tests to see people have become conditioned to skip through onboarding wizards and modals like these. A more native embedded approach would be a better way to go in my opinion.

scruffian commented:
Ok in that case I think we also need a companion PR to remove it from Atomic sites. I'll work on that now

scruffian committed:
Remove the What's New app

tyxla reviewed (commented):
Note that once since we're removing the app, there is one last usage of the `@automattic/whats-new` package which we might be able to remove, and delete the package altogether.

scruffian committed:
Remove other uses of What's New

tyxla reviewed (commented):
Let's also remove the package (this is the primary reason for the CI failures) and consider following up to remove the app from WPCOM altogether.

scruffian committed:
rebuild

fditrapani commented:
In case you haven't already got this, I just noticed we have this in the Editor as well:
<img width="1840" alt="image" src="https://github.com/user-attachments/assets/cfc9c9e7-7d49-4db0-ac31-12f5a5efb968" />

fditrapani commented:
Yeah, I think so too. We'll want to remove the menu item just under help.

scruffian commented:
Ah I hadn't seen that!

escapemanuele commented:
In case you haven't seen it, that link in the menu should come from [here](https://github.com/Automattic/wp-calypso/blob/trunk/apps/whats-new/src/index.js) @scruffian

scruffian committed:
undo height change

scruffian committed:
remove whats new logic from the help icon

scruffian committed:
Remove  from the store

scruffian committed:
remove hasSeenWhatsNewModal

scruffian commented:
Ok, I have updated the HelpIcon to not use `getHasSeenWhatsNewModal`. I have also removed `getHasSeenWhatsNewModal` from the store.

There is still a `whats-new` package. I assume that can be removed once we merge https://github.com/Automattic/jetpack/pull/42672/?

I'm inclined to do that in a different PR, so that we can deploy this one and that Jetpack one together, removing the option in both places at the same time. Then we can do a followup PR to remove the package. Does that make sense @tyxla?

fditrapani reviewed (commented):
Thanks for the updates. I noticed something off with when the help opens now:


https://github.com/user-attachments/assets/1e5db689-4b99-412c-bc8c-62b530268ee1

It feels too short and scrolls to the bottom every time.

scruffian commented:
@filipanoscampos do you have the latest changes? At one point I changed the height of the Help Centre, but I reverted it later on in the PR. I think you get the latest version you shouldn't see that issue anymore.

fditrapani commented:
I think you meant that last comment for me. My branch was stale. Just updating everything and testing again.

fditrapani reviewed (approved):
Looks like I did have a stale PR. This looks good on my end after updating.

scruffian committed:
Remove the what's new package

scruffian committed:
update lock file

tyxla commented:
> Actually looking at this again with fresh eyes I think we can also remove the package in the PR too. I've done that and updated the lock file.

Great to hear! I'll try to have a look at it today.

tyxla commented on packages/data-stores/src/help-center/actions.ts:
```diff
@@ -16,37 +14,6 @@ import type {
 } from './types';
 import type { SupportInteraction } from '@automattic/odie-client/src/types';
 
-export const receiveHasSeenWhatsNewModal = ( value: boolean | undefined ) =>
```
Looks like we need to adjust some types to get the builds to pass:

https://github.com/Automattic/wp-calypso/blob/b9bac06a4d9649a52ad98b7228f8724b3704a500/packages/data-stores/src/help-center/actions.ts#L225

scruffian replied:
Thanks? How did you find the error?

tyxla replied:
TC shows it in the failed build: https://teamcity.a8c.com/buildConfiguration/calypso_BuildDockerImage/14611117?expandBuildDeploymentsSection=false&hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&showLog=14611117_1053_756&logView=flowAware

tyxla replied:
Also, if you run `yarn run postinstall` locally, it should show it. Here's what I saw when running on your branch (before you pushed the [last commit](https://github.com/Automattic/wp-calypso/pull/101692/commits/c8c8ec7dcdb12bddf56dbc3ed11912854fa6a98c)):

![Screenshot 2025-03-27 at 15 38 56](https://github.com/user-attachments/assets/1f6e6e15-b2ad-4417-abb5-673a10aa8fae)

tyxla reviewed (changes requested):
`yarn install` (`postinstall` specifically) fails right now. Seems like it's caused by type errors, see inline comment.

scruffian committed:
remove usused type definition

tyxla reviewed (commented):
Seems like in packages we still try to load the TS config of the removed package:

https://github.com/Automattic/wp-calypso/blob/bcc160d2b4f34d87c2f20c176db93409199a41b2/packages/tsconfig.json#L59

There may be more occurrences to remove, you can find any suspects by looking at `whats-new` in the codebase.

I'd recommend running locally and making sure that you can:
* Run `yarn run distclean` to start fresh
* Run `yarn install` without issues
* Build Calypso without issues

Happy to review more thoroughly once it's all good and green and ready to test.

scruffian committed:
remove more uses of the whats-new package

scruffian committed:
update e2e test

tyxla commented on apps/whats-new/package.json:
```diff
@@ -1,59 +0,0 @@
-{
-	"name": "@automattic/whats-new-app",
```
If we're removing the `whats-new` app, let's also remove its artifact from the WP.com plugins build:

https://github.com/Automattic/wp-calypso/blob/7cfe55f34a4af56b2c066e8ed93bb9b9889a5365/.teamcity/_self/projects/WPComPlugins.kt#L46

https://github.com/Automattic/wp-calypso/blob/7cfe55f34a4af56b2c066e8ed93bb9b9889a5365/.teamcity/_self/projects/WPComPlugins.kt#L129

tyxla commented on apps/whats-new/src/index.js:
```diff
@@ -1,51 +0,0 @@
-	return (
-		<>
-			<Fill name="ToolsMoreMenuGroup">
-				<MenuItem onClick={ openWhatsNew }>{ __( "What's new", 'full-site-editing' ) }</MenuItem>
```
This looks like the wrong textdomain 😅 were these translations even working?

tyxla commented on packages/data-stores/src/help-center/actions.ts:
```diff
@@ -16,37 +14,6 @@ import type {
-	} else {
-		response = yield apiFetch( {
-			global: true,
-			path: '/wpcom/v2/block-editor/has-seen-whats-new-modal',
```
Can/should some of these endpoints also be cleaned up?

scruffian replied:
I guess they'll still be called in the desktop app for a while. I retired the old themes endpoint about a month ago but it's still getting some hits. I'll make a note of it and watch what the numbers do.

tyxla commented on packages/help-center/src/components/help-center-more-resources.tsx:
```diff
@@ -117,31 +69,7 @@ export const HelpCenterMoreResources = () => {
-							<Button
-								variant="link"
-								onClick={ handleWhatsNewClick }
-								className="help-center-more-resources__new-releases"
```
There are some unused styles to be removed related to this no-longer-used classname.

tyxla commented on packages/help-center/src/components/help-center-more-resources.tsx:
```diff
@@ -2,55 +2,15 @@
 import { Icon, formatListNumbered, external, institution } from '@wordpress/icons';
 import { useI18n } from '@wordpress/react-i18n';
 import { useHelpCenterContext } from '../contexts/HelpCenterContext';
-import { NewReleases } from '../icons';
```
This icon doesn't seem to be used anymore. Let's remove it as well.

tyxla commented on packages/help-center/src/components/help-center-more-resources.tsx:
```diff
@@ -117,31 +69,7 @@ export const HelpCenterMoreResources = () => {
-								<span>{ __( "What's New", __i18n_text_domain__ ) }</span>
-								{ showWhatsNewDot && (
-									<Icon
-										className="help-center-more-resources__new-releases_dot"
```
There are some unused existing styles on this classname that we can clean up.

tyxla commented on packages/whats-new/src/components/guide.tsx:
```diff
@@ -1,95 +0,0 @@
-/**
- * WordPress dependencies
- */
-import { PaginationControl } from '@automattic/components';
```
This is the last usage of this `PaginationControl` component. Let's remove it.

tyxla commented on packages/whats-new/src/style.scss:
```diff
@@ -1,272 +0,0 @@
-@import "@automattic/typography/styles/fonts";
-@import "@automattic/onboarding/styles/mixins";
-@import "@automattic/calypso-color-schemes";
```
Another color schemes import down (cc @jasmussen for sharing the celebration)

jasmussen replied:
Hooray!

tyxla reviewed (commented):
This is great work, thank you @scruffian 🙌 

I think we should go ahead and land this one: https://github.com/Automattic/jetpack/pull/42672

Also, I've left a few more opportunities to clean up this PR and remove more code. Let me know what you think.

scruffian committed:
remove artifacts from the build

scruffian committed:
Remove unused CSS

scruffian committed:
Remove unused icon

scruffian committed:
remove PaginationControl

scruffian committed:
remove Icon includes

tyxla reviewed (approved):
This is looking great! ✅ 

Thanks @scruffian for the great cleanup!

🚀
